File: backend/src/main/java/top/vuhe/common/ApiResponseSerializer.java
package top.vuhe.common;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.SerializerProvider;
import com.fasterxml.jackson.databind.ser.std.StdSerializer;

import java.io.IOException;

/**
 * 序列化 通用返回类
 *
 * @author zhuhe
 */
public class ApiResponseSerializer extends StdSerializer<ApiResponse<?>> {
    public ApiResponseSerializer() {
        this(null);
    }

    public ApiResponseSerializer(Class<ApiResponse<?>> t) {
        super(t);
    }

    @Override
    public void serialize(ApiResponse<?> value, JsonGenerator gen, SerializerProvider provider) throws IOException {
        gen.writeStartObject();
        gen.writeNumberField("code", value.getCode());
        gen.writeStringField("message", value.getMessage());
        if (value.getData() != null) {
            gen.writeObjectField(value.getDataName(), value.getData());
        }
        gen.writeEndObject();
    }
}

File: backend-kt/src/main/kotlin/top/vuhe/sw/common/ApiResponseSerializer.kt
package top.vuhe.sw.common

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.SerializerProvider
import com.fasterxml.jackson.databind.ser.std.StdSerializer

/**
 * 序列化 通用返回类
 *
 * @author zhuhe
 */
class ApiResponseSerializer : StdSerializer<ApiResponse<*>> {

    constructor() : this(null)

    constructor(t: Class<ApiResponse<*>>?) : super(t)

    override fun serialize(
        value: ApiResponse<*>,
        gen: JsonGenerator,
        provider: SerializerProvider?
    ) {
        gen.writeStartObject()
        gen.writeNumberField("code", value.code)
        gen.writeStringField("message", value.message)
        if (value.data != null) {
            gen.writeObjectField("data", value.data)
        }
        gen.writeEndObject()
    }

}